sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/async_challenge_auth_policy.py

```diff
@@ -55,3 +79,19 @@ class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
 
         access_token = await self._credential.get_token(scope)
         self._update_headers(request.http_request.headers, access_token.token)
+
+    @staticmethod
+    def _log_request(request):
+        header_string = ""
+        for header in request.http_request.headers:
+            header_string += (", " + header)
```

Comment:
`", ".join(request.http_request.headers)`

Reply:
fixed, thanks that's a lot cleaner and I'll be sure to remember that next time!